### PR TITLE
Small Pythonic code changes for datastructures.py file

### PR DESCRIPTION
**Pull request by ShahriyarR**
The aim is to simplify and add more style :)
Thanks.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,6 @@ venv*/
 .python-version
 build/
 dist/
+.idea/
```
Comment (Kludex) [inline .gitignore]: Those ignores are not going to be accepted :sweat_smile:

Comment (Kludex) [reply .gitignore]: I tried: https://github.com/encode/uvicorn/pull/1001

Comment (ShahriyarR) [reply .gitignore]: ahahaha :D Okay reverting back)

Comment (ShahriyarR) [reply .gitignore]: Done

File: .gitignore [hunk 2]
```diff
@@ -10,4 +10,4 @@ site/
 venv*/
 .python-version
 build/
-dist/
+dist/
```
Comment (Arlington1985) [inline .gitignore]: Why remove the new line at the end? :)

**Review by fuadaghazada — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (ShahriyarR) [reply .gitignore]: :) done

File: starlette/datastructures.py [hunk 3]
```diff
@@ -584,10 +576,11 @@ def __setitem__(self, key: str, value: str) -> None:
         set_key = key.lower().encode("latin-1")
         set_value = value.encode("latin-1")
 
-        found_indexes = []
-        for idx, (item_key, item_value) in enumerate(self._list):
-            if item_key == set_key:
-                found_indexes.append(idx)
+        found_indexes = [
```
Comment (Kludex) [inline starlette/datastructures.py]: This is controversial. Not everybody believes it's an improvement. I'd prefer to not change it.

File: starlette/datastructures.py [hunk 4]
```diff
@@ -604,10 +597,11 @@ def __delitem__(self, key: str) -> None:
         """
         del_key = key.lower().encode("latin-1")
 
-        pop_indexes = []
-        for idx, (item_key, item_value) in enumerate(self._list):
-            if item_key == del_key:
-                pop_indexes.append(idx)
+        pop_indexes = [
```
Comment (Kludex) [inline starlette/datastructures.py]: Same here.

**Review by Kludex — CHANGES_REQUESTED** [review]
@ShahriyarR  Thanks for the PR! :tada: 

The two first changes I think it does improve readability. But I don't feel the same for the last two. Can you check my comments?

**Review by Kludex — APPROVED** [review]
Thanks for the PR @ShahriyarR ! :tada: 

:)

**Comment by ShahriyarR** [comment]
@Kludex awesome merge :D

**Pull request merged**